### DISCUSSION TITLE
Output of other forms of atomic charge

### DISCUSSION THREAD

**Discussion by scott-5** [deepmodeling/abacus-develop#2497] [Ideas]
In addition to outputting mulliken charges (which in some cases may not be described accurately enough), would ABACUS consider outputting other forms of atomic charges (like Hirshfeld and so on), or take another option: outputting wavefunction files in e.g. .molden format, through which the user can analyze atomic charges using third-party analysis software (Multiwfn).

**Comment by Satinelamp** [comment]
Maybe you could share the algorithm or paper about new forms of atomic charge.

**Reply by scott-5** [reply]
Natural population analysis:
J. Chem. Theory Comput., 2012, 8, 1902
https://github.com/jrschmidt2/periodic-NBO

Hirshfeld population:
Theoret. Chim. Acta, 1977, 44, 129
J. Theor. Comput. Chem., 2012, 11, 163
https://github.com/frobnitzem/hirshfeld/tree/main/pyscf/hirshfeld

not atomic charge but popular analytical methods:
J. Phys. Chem., 1993, 97, 8617
J. Phys. Chem. A, 2011, 115, 5461
